distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1543,6 +1543,196 @@ public void testFederatedDownloadProductToCacheOnlyCacheDisabled() throws Except
+
+        validateNotificationsFailure(cometDClient1, filename);
+        validateNotificationsFailure(cometDClient2, filename);
+
+        validateActivitiesFailure(cometDClient1, filename);
+        validateActivitiesFailure(cometDClient2, filename);
+
+    }
+
+    private void setupCswServer(String metacardId, String filename, String resourceData)
+            throws Exception {
+        cswServer.whenHttp()
+                .match(post("/services/csw"),
+                        withPostBodyContaining("GetRecords"),
+                        withPostBodyContaining(metacardId))
+                .then(ok(),
+                        contentType("text/xml"),
+                        bytesContent(getCswQueryResponse(metacardId).getBytes()));
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId))
+                .then(getCswRetrievalHeaders(filename),
+                        chunkedContentWithHeaders(resourceData, Duration.ofMillis(200), 3));
+
+    }
+
+    private void checkExpectations(CometDClient cometDClient, Integer numNotifications,
+            Integer numActivities) throws Exception {
+        expect("Waiting for notifications").within(10, SECONDS)
+                .until(() -> cometDClient.getMessages(NOTIFICATIONS_CHANNEL)
+                        .size() == numNotifications);
+        expect("Waiting for activities").within(10, SECONDS)
+                .until(() -> cometDClient.getMessages(ACTIVITIES_CHANNEL)
+                        .size() == numActivities);
+
+    }
+
+    private void validateNotificationsSuccess(CometDClient cometDClient, String filename,
+            String resourceData) throws Exception {
+
+        List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(1));
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)),
+                filename,
+                getResourceRetrievalCompletedMessage(resourceData.length()),
+                "complete");
+
+    }
+
+    private void validateActivitiesSuccess(CometDClient cometDClient, String filename,
+            String resourceData) throws Exception {
+        List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(1));
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)),
+                filename,
+                getResourceRetrievalCompletedMessage(resourceData.length()),
+                "complete");
+
+    }
+
+    private void validateNotificationsFailure(CometDClient cometDClient, String filename)
+            throws Exception {
+        List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(4));
```

Comment:
If you’re looking to determine if the notification indicated a failure in “validateNotificationsFailure”, the cometDClient’s notifications or activities channel can be stored as a List\<String\> and then checked if it contains the word “failed.” Not sure if that’s a better or worse way to do it, but it might mean you wouldn’t need to check for a 4 notifications size. This is how I ended up doing it in the “foundExpectedActivity” helper method here: https://github.com/codice/ddf/pull/1009/files